pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -93,8 +93,10 @@ public class ForwardIndexHandler extends BaseIndexHandler {
   private final Schema _schema;
 
   protected enum Operation {
-    DISABLE_FORWARD_INDEX_FOR_DICT_COLUMN,
-    DISABLE_FORWARD_INDEX_FOR_RAW_COLUMN,
+    DISABLE_FORWARD_INDEX_FOR_DICT_OR_RAW_COLUMN,
```

Comment:
This part is very complicated. Are we able to simplify the operations allowed?
I feel we can simplify it by making changes in steps:
1. Check if we need to enable/disable dictionary (switch between dictionary encoding and raw forward index)
2. Check if we need to enable/disable the forward index
3. Check if we need to change the raw index compression type

Reply:
Let me look at the code and get back to you about the feasibility of doing the above in steps

Reply:
@Jackie-Jiang I was able to take your suggestion and simplify the `ENUMS` used here and perform the operations in steps. Can you take a look and let me know what you think? Thanks!